plugin/src/main/java/org/opensearch/ml/helper/ConnectorAccessControlHelper.java

```diff
@@ -0,0 +1,176 @@
+/*
+ *
+ *  * Copyright OpenSearch Contributors
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.opensearch.ml.helper;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.ml.settings.MLCommonsSettings.ML_COMMONS_CONNECTOR_ACCESS_CONTROL_ENABLED;
+
+import org.apache.lucene.search.join.ScoreMode;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.get.GetRequest;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.CollectionUtils;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.commons.authuser.User;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.NestedQueryBuilder;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.index.query.TermQueryBuilder;
+import org.opensearch.ml.common.AccessMode;
+import org.opensearch.ml.common.CommonValue;
+import org.opensearch.ml.common.connector.AbstractConnector;
+import org.opensearch.ml.common.connector.Connector;
+import org.opensearch.ml.common.exception.MLResourceNotFoundException;
+import org.opensearch.ml.utils.MLNodeUtils;
+import org.opensearch.ml.utils.RestActionUtils;
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class ConnectorAccessControlHelper {
```

Comment:
This class already included in Zan's PR: https://github.com/opensearch-project/ml-commons/pull/1055

Reply:
The build is failing without this class. So let's merge this 1055 PR first.